tests/v2/validation/provisioning/rke2/etcd_backup_restore_test.go

```diff
@@ -56,612 +41,117 @@ func (r *RKE2EtcdSnapshotRestoreTestSuite) SetupSuite() {
-	require.Equal(r.T(), deploymentAfterBackup.Name, deploymentResp.ObjectMeta.Name)
-	logrus.Infof("%v is ready.............", deploymentAfterBackup.Name)
-
-	logrus.Infof("upgrading cluster k8s version.............")
-	k8sUpgradedVersion := r.kubernetesVersions[1]
-	err = upgradeClusterK8sVersionWithUpgradeStrategy(client, clusterName, k8sUpgradedVersion, r.ns)
-	require.NoError(r.T(), err)
-	clusters.WatchAndWaitForCluster(r.client.Steve, kubeProvisioningClient, r.ns, clusterName)
-	logrus.Infof("cluster is active again.............")
-
-	cluster, _, err := clusters.GetProvisioningClusterByName(client, clusterName, r.ns)
-	require.NoError(r.T(), err)
-	require.Equal(r.T(), k8sUpgradedVersion, cluster.Spec.KubernetesVersion)
-
-	logrus.Infof("validating ControlPlaneConcurrency and WorkerConcurrency values are updated..")
-	require.Equal(r.T(), "15%", cluster.Spec.RKEConfig.UpgradeStrategy.ControlPlaneConcurrency)
-	require.Equal(r.T(), "20%", cluster.Spec.RKEConfig.UpgradeStrategy.WorkerConcurrency)
-
-	logrus.Infof("restoring snapshot.............")
-	require.NoError(r.T(), restoreSnapshot(client, clusterName, snapshotToBeRestored, 1, "all", r.ns))
-	logrus.Infof("successfully submitted restoration request.............")
-
-	logrus.Infof("creating watch over cluster after restore.............")
-	clusters.WatchAndWaitForCluster(r.client.Steve, kubeProvisioningClient, r.ns, clusterName)
-	logrus.Infof("cluster is active again.............")
-
-	logrus.Infof("creating watch over pods.............")
-	r.watchAndWaitForPods(client, clusterID)
-	logrus.Infof("All pods are up and running.............")
-
-	logrus.Infof("fetching deployment list to validate restore.............")
-	deploymentList, err := steveclient.SteveType(workloads.DeploymentSteveType).NamespacedSteveClient(r.ns).List(nil)
-	require.NoError(r.T(), err)
-	require.Equal(r.T(), 1, len(deploymentList.Data))
-	require.Equal(r.T(), wloadBeforeRestore, deploymentList.Data[0].ObjectMeta.Name)
-	logrus.Infof(" deployment list validated successfully.............")
-
-	logrus.Infof("fetching ingresses list to validate restore.............")
-	ingressResp, err = steveclient.SteveType(ingresses.IngressSteveType).ByID(r.ns + "/" + ingressBeforeBackup.Name)
-	require.NoError(r.T(), err)
-	require.NotNil(r.T(), ingressResp)
-	logrus.Infof("ingress validated successfully.............")
-
-	cluster, _, err = clusters.GetProvisioningClusterByName(client, clusterName, r.ns)
+func (r *RKE2EtcdSnapshotRestoreTestSuite) TestEtcdSnapshotRestoreWithK8sUpgrade() {
+	logrus.Infof("checking for valid cnis in the configuration....")
+	require.GreaterOrEqual(r.T(), len(r.clustersConfig.Basics.CNIs), 1)
+	nodeRoles := []machinepools.NodeRoles{
+		{
+			ControlPlane: true,
+			Quantity:     2,
+		},
+		{
+			Etcd:     true,
+			Quantity: 3,
+		},
+		{
+			Worker:   true,
+			Quantity: 3,
+		},
+	}
+	testSession := session.NewSession()
+	defer testSession.Cleanup()
+	_, err := r.client.WithSession(testSession)
```

Comment:
Ignoring the client returning, means this isn't doing what you expect. It's completely ignoring the subsession did you mean to do this?

Reply:
the subsession is created down the line, let me remove this.